src/badguy/goldbomb.cpp

```diff
@@ -139,15 +160,108 @@ GoldBomb::active_update(float dt_sec)
     }
     return;
   }
-  if (is_grabbed())
+
+  if ((tstate == STATE_FLEEING || tstate == STATE_CORNERED) && on_ground() && might_fall(FLEEING_MAX_DROP_HEIGHT+1))
+  {
+    // also check for STATE_CORNERED just so
+    // the bomb doesnt automatically turn around
+    cornered();
     return;
+  }
   WalkingBadguy::active_update(dt_sec);
+
+  if (m_frozen) return;
+
+  MovingObject* obj = nullptr;
+  std::vector<MovingObject*> objs = Sector::get().get_nearby_objects(get_bbox().get_middle(), SAFE_DIST);
+  for (MovingObject* currobj : objs)
+  {
+    obj = currobj;
```

Comment:
Why is the extra variable needed?

Reply:
It's the object that the gold bomb will check. read the rest of the code if you haven't

Reply:
Can you not use the loop iterator variable for that?

Reply:
Will it stay in scope?

Reply:
You're not using this anywhere else except for the for loop, no?

Reply:
I do use this outside the loop

Reply:
Ah, sorry, the code logic was...copulating with my brain.

Reply:
i think it should also be tweaked. any suggestions?

Reply:
@tobbi this is resolved is it?